### PR TITLE
add documentation

### PR DESCRIPTION
**Pull request by thibaultrey**

### CODE REVIEW

**Comment by thibaultrey** [comment]
@pierrickbrun

File: data/doc_pages/en/cart.md [hunk 1]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
 |--------|--------|-------------|
+| amount | Objects | Object with shippings cost taxed and without VAT |
+| selected_carrier | Objects | Current selected carrier |
+| address | Objects | Customer shipping address |
 
 
+
+> Make an API request to cart/get_delivery_methods to get available carrier list according to delivery area.
+>
+> **Selected_carrier** and **amount are** set only after call cart/apply_delivery_method API method
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
> **Selected_carrier** and **amount** are set only after call cart/apply_delivery_method API method
```

File: data/doc_pages/en/cart.md [hunk 2]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+Invoicing customer address, selected carrier and shipping cost.
+
+| Name | Type | Description |
+|--------|--------|-------------|
+| address | Objects | Customer invoicing address |
+
+### Lines
+
+This attribute is a collection of cart lines. 
+Line object have to nested attributes :
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
Line object have two nested attributes :
```

File: data/doc_pages/en/cart.md [hunk 3]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+| selected_method | Object | selected payment method by the customer  |
+| amount | Float | grand total  |
+| selected_method | Objects | List of all available payment method  |
 
 ## Methods
 
+### [POST] cart/update
+
+Update cart data. This method is used to change shipping or invoicing address, add note, set a new cart step.
+cart step is used to get cart feedback for marchant cart analysis. It's also used to transform a cart to an order after the checkout.
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
cart step is used to get cart feedback for marchant cart analysis. It's also used to transform a cart into an order after the checkout.
```

File: data/doc_pages/en/cart.md [hunk 4]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+| shipping.address.id | Integer | Id of selected shipping address, optionnal value. See Addresses Documentation to get customer addresses list  |
+| invoicing.address.id | Integer | Id of selected invoicing address, optionnal value. See Addresses Documentation to get customer addresses list  |
+
+invoicing.address.id and shipping.address.id are not required values.
+
+#### Example
+
+Cart update via simple HTML form. 
+This example is a form to set a shipping address in cart. 
+Set a shipping address is necessary to make API call on `cart/get_delivery_methods` to get carrier list.
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
Setting a shipping address is necessary to make API calls on `cart/get_delivery_methods` to get the carrier list. 
```

File: data/doc_pages/en/cart.md [hunk 5]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+| invoicing.address.id | Integer | Id of selected invoicing address, optionnal value. See Addresses Documentation to get customer addresses list  |
+
+invoicing.address.id and shipping.address.id are not required values.
+
+#### Example
+
+Cart update via simple HTML form. 
+This example is a form to set a shipping address in cart. 
+Set a shipping address is necessary to make API call on `cart/get_delivery_methods` to get carrier list. 
+`invader_success_url` and `invader_error_url` params are used for HTTP redirection after submition according to the response state (valid or error).
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
`invader_success_url` and `invader_error_url` parameters are used for HTTP redirection after submission according to the response state (valid or error).
```

File: data/doc_pages/en/cart.md [hunk 6]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+      {% for address in store.addresses %}
+        <label>
+          <input type="radio" value="{{address.id | round}}" name="shipping[address][id]" />
+          {{ address.name }}
+        </label>
+      {% endfor %}
+      <input type="submit" name="submit">
+    </form>
+
+_To see how to call API method form HTML From See [Call API from HTML Form](./docs/api-call-in-form)._
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
_To see how to call API method from a HTML form See [Call API from HTML Form](./docs/api-call-in-form)._
```

File: data/doc_pages/en/cart.md [hunk 7]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+
+
+### [POST] Method cart/add_item 
+
+Add product to cart. This method create a new line in cart and refresh `store.cart.lines` variable.
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
Add product to cart. This method creates a new line in the cart and refreshes the `store.cart.lines` variable.
```

File: data/doc_pages/en/cart.md [hunk 8]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+| step.next | String | New cart step  |
+| shipping.address.id | Integer | Id of selected shipping address, optionnal value. See Addresses Documentation to get customer addresses list  |
+| invoicing.address.id | Integer | Id of selected invoicing address, optionnal value. See Addresses Documentation to get customer addresses list  |
+
+invoicing.address.id and shipping.address.id are not required values.
+
+#### Example
+
+Cart update via simple HTML form. 
+This example is a form to set a shipping address in cart.
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
This example is a form to set a shipping address in the cart. 
```

File: data/doc_pages/en/cart.md [hunk 9]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+#### Options
+
+| Name | Type | Description |
+|--------|--------|-------------|
+| product_id | Integer | Product objectID  |
+| item_qty | Integer | Number of product added in cart  |
+
+#### Example
+
+Add new item in cart via simple HTML form.
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
Add a new item in the cart via a simple HTML form. 
```

File: data/doc_pages/en/cart.md [hunk 10]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+
+| Name | Type | Description |
+|--------|--------|-------------|
+| product_id | Integer | Product objectID  |
+| item_qty | Integer | Number of product added in cart  |
+
+#### Example
+
+Add new item in cart via simple HTML form. 
+This form add 10 product with ObjectID "123456" in the cart.
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
This form adds 10 products with ObjectID "123456" in the cart.
```

File: data/doc_pages/en/cart.md [hunk 11]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+
+    <form method="post" action="/invader/cart/add_item">
+      <input type="hidden" name="invader_success_url" value="/my-cart" />
+      <input type="hidden" name="invader_error_url" value="/my-error-page" />
+      <input type="hidden" name="item_qty" value="10">
+      <input type="hidden" name="product_id" value="123456">
+      <input type="submit" name="submit">
+    </form>
+
+_To see how to call API method form HTML From See [Call API from HTML Form](./docs/api-call-in-form)._
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
_To see how to call API method from a HTML form See [Call API from HTML Form](./docs/api-call-in-form)._
```

File: data/doc_pages/en/cart.md [hunk 12]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+
+### [POST] Method cart/update_item
+
+Update cart line quantity.
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
Update a cart line quantity.
```

File: data/doc_pages/en/cart.md [hunk 13]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+#### Options
+
+| Name | Type | Description |
+|--------|--------|-------------|
+| item_id | Integer | cart line ID  |
+| item_qty | Integer | New quantity  |
+
+#### Example
+
+Add new item in cart via simple HTML form.
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
Add a new item in the cart via a simple HTML form. 
```

File: data/doc_pages/en/cart.md [hunk 14]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+
+| Name | Type | Description |
+|--------|--------|-------------|
+| item_id | Integer | cart line ID  |
+| item_qty | Integer | New quantity  |
+
+#### Example
+
+Add new item in cart via simple HTML form. 
+This form update line #1 to set quantity 8.
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
This form updates line #1 to set quantity = 8.
```

File: data/doc_pages/en/cart.md [hunk 15]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+
+    <form method="post" action="/invader/cart/update_item">
+      <input type="hidden" name="invader_success_url" value="/my-cart" />
+      <input type="hidden" name="invader_error_url" value="/my-error-page" />
+      <input type="hidden" name="item_qty" value="8">
+      <input type="hidden" name="item_id" value="1">
+      <input type="submit" name="submit">
+    </form>
+
+_To see how to call API method form HTML From See [Call API from HTML Form](./docs/api-call-in-form)._
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
_To see how to call API method from a HTML form See [Call API from HTML Form](./docs/api-call-in-form)._
```

File: data/doc_pages/en/cart.md [hunk 16]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+
+### [GET] Method cart/get_delivery_methods
+
+Get the list of all delivery method available according to cart data (shipping area).
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
Get the list of all delivery methods available according to cart data (shipping area).
```

File: data/doc_pages/en/cart.md [hunk 17]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+      </label>
+    {% endfor %}
+
+
+_To see how to get data from  API see [ERP tag on Special tag ](./docs/special-tag)._
+
+
+### [POST] Method cart/apply_delivery_method
+
+Set selected carrier on cart. This method causes shipping cost calculation depends on carrier, cart content and shipping area.
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
Set selected carrier on cart. This method causes shipping cost calculation depending on carrier, cart content and shipping area.
```

File: data/doc_pages/en/cart.md [hunk 18]
```diff
@@ -38,24 +38,210 @@ Amout object provide cart Grand total and taxes amount.
...
+      {% for carrier in delivery_methods %}
+        <label>
+          <input type="radio" value="{{carrier.id | round}}" name="carrier[id]" />
+          {{ carrier.name }}
+        </label>
+      {% endfor %}
+      <input type="submit" name="submit">
+    </form>
+
+_To see how to call API method form HTML From See [Call API from HTML Form](./docs/api-call-in-form)._
```
Comment (PierrickBrun) [inline data/doc_pages/en/cart.md]:
```suggestion
_To see how to call API method from a HTML form See [Call API from HTML Form](./docs/api-call-in-form)._
```

File: public/javascripts/document-menu.js [hunk 19]
```diff
@@ -20,6 +25,7 @@ $(document).ready(function () {
 });
 
 $(document).ready(function () {
+	/*
```
Comment (PierrickBrun) [inline public/javascripts/document-menu.js]: to delete

**Review by PierrickBrun — CHANGES_REQUESTED** [review]

**Pull request merged**